extensions/android-paging/src/main/java/com/squareup/sqldelight/android/paging/QueryDataSourceFactory.kt

```diff
@@ -3,18 +3,34 @@ package com.squareup.sqldelight.android.paging
 import androidx.paging.DataSource
 import androidx.paging.PositionalDataSource
 import com.squareup.sqldelight.Query
+import com.squareup.sqldelight.Transacter
 
 class QueryDataSourceFactory<RowType : Any>(
   private val queryProvider: (limit: Long, offset: Long) -> Query<RowType>,
-  private val countQuery: Query<Long>
+  private val countQuery: Query<Long>,
+  private val transacter: Transacter?
```

Comment:
lets just update the constructor, i'm not concerned about ABI compatibility for the paging component

Reply:
plus then this doesnt need to be nullable

Reply:
I'm down, more so cause the implementation is buggy without the `Transacter`.

Reply:
Updated the PR. Probably worth calling out in the CHANGELOG that this breaks compat.

Reply:
definitely